nym-vpn-core/crates/nym-vpn-account-controller/src/controller.rs

```diff
@@ -107,13 +108,21 @@ where
             storage_op_sender,
         );
 
-        let (current_state_handler, initial_state) = if shared_state
+        let is_offline = shared_state
             .connectivity_handle
             .connectivity()
             .await
-            .is_offline()
-        {
+            .is_offline();
+        let is_in_decentralised_mode = shared_state
```

Comment:
Since you added the proper transition at the start of syncing mode, this adds unnecessary complexity IMO. You can revert back to Offline/Syncing initial state